Mathlib/CategoryTheory/FiberedCategory/Cartesian.lean

```diff
@@ -115,4 +143,200 @@ instance of_comp_iso {b' : 𝒳} (φ' : b ≅ b') [IsHomLift p (𝟙 S) φ'.hom]
     apply map_uniq
     simp only [Iso.eq_comp_inv, assoc, hτ₂]
 
-end Functor.IsCartesian
+end IsCartesian
+
+namespace IsStronglyCartesian
+
+section
+
+variable {R S : 𝒮} {a b : 𝒳} (f : R ⟶ S) (φ : a ⟶ b) [IsStronglyCartesian p f φ]
+
+/-- The universal property of a strongly cartesian morphism.
+
+This lemma is more flexible with respect to non-definitional equalities than the field
+`universal_property'` of `IsStronglyCartesian`. -/
+lemma universal_property {R' : 𝒮} {a' : 𝒳} (g : R' ⟶ R) (f' : R' ⟶ S) (hf' : f' = g ≫ f)
+    (φ' : a' ⟶ b) [IsHomLift p f' φ'] : ∃! χ : a' ⟶ a, IsHomLift p g χ ∧ χ ≫ φ = φ' := by
+  subst_hom_lift p f' φ'; clear a b R S
+  have : p.IsHomLift (g ≫ f) φ' := (hf' ▸ inferInstance)
+  apply IsStronglyCartesian.universal_property' f
+
+instance isCartesian_of_isStronglyCartesian [p.IsStronglyCartesian f φ] : p.IsCartesian f φ where
+  universal_property := fun φ' => universal_property p f φ (𝟙 R) f (by simp) φ'
+
+section
+
+variable {R' : 𝒮} {a' : 𝒳} {g : R' ⟶ R} {f' : R' ⟶ S} (hf' : f' = g ≫ f) (φ' : a' ⟶ b)
+  [IsHomLift p f' φ']
+
+/-- Given a diagram
+```
+a'        a --φ--> b
+|         |        |
+v         v        v
+R' --g--> R --f--> S
+```
+such that `φ` is a cartesian, and a morphism `φ' : a' ⟶ b`. Then `map` is the map `a' ⟶ a`
+lying over `g` obtained from the universal property of `φ`. -/
+noncomputable def map : a' ⟶ a :=
+  Classical.choose <| universal_property p f φ _ _ hf' φ'
+
+instance map_isHomLift : IsHomLift p g (map p f φ hf' φ') :=
+  (Classical.choose_spec <| universal_property p f φ _ _ hf' φ').1.1
+
+@[simp]
+lemma fac : (map p f φ hf' φ') ≫ φ = φ' :=
+  (Classical.choose_spec <| universal_property p f φ _ _ hf' φ').1.2
+
+/-- Given a diagram
+```
+a'        a --φ--> b
+|         |        |
+v         v        v
+R' --g--> R --f--> S
+```
+such that `φ` is a cartesian, and morphisms`φ' : a' ⟶ b` and `ψ : a' ⟶ a` such that `g ≫ ψ = φ'`.
+Then `ψ` is the map induced by the universal property. -/
+lemma map_uniq (ψ : a' ⟶ a) [IsHomLift p g ψ] (hψ : ψ ≫ φ = φ') : ψ = map p f φ hf' φ' :=
+  (Classical.choose_spec <| universal_property p f φ _ _ hf' φ').2 ψ ⟨inferInstance, hψ⟩
+
+/-- Given a diagram
+```
+a'        a --φ--> b
+|         |        |
+v         v        v
+R' --g--> R --f--> S
+```
+such that `φ` is cartesian, and morphisms `φ' : a' ⟶ b`, `ψ ψ' : a' ⟶ a` such that
+`g ≫ ψ = φ' = g ≫ ψ'`. Then `ψ = ψ'`. -/
+protected lemma eq_of_fac {ψ ψ' : a' ⟶ a} [IsHomLift p g ψ] [IsHomLift p g ψ']
```

Comment:
Is it possible to formulate this lemma without referring to `φ'` so that the assumption become `ψ ≫ φ = ψ' ≫ φ`? If so, the lemma could be renamed `ext`?

Reply:
Great suggestion!